salt/modules/junos.py

```diff
@@ -1079,3 +1097,186 @@ def file_copy(src=None, dest=None):
         ret['message'] = 'Could not copy file : "{0}"'.format(exception)
         ret['out'] = False
     return ret
+
+
+def lock():
+    """
+    Attempts an exclusive lock on the candidate configuration. This
+    is a non-blocking call.
+
+    .. note::
+        Any user who wishes to use lock, must necessarily unlock the
+        configuration too.
```

Comment:
Not only must the user call unlock, they should make sure they call unlock in the same orchestration run in which the lock was called.